### PR TITLE
Index pattern on the filters of home KPI links

### LINKED ISSUE #6776: Home overview KPI's link don't work with custom index-pattern
| Wazuh | Rev  | Browser                      |
|-------|------|------------------------------|
| 4.8.0 | 12 | Chrome, Firefox, Safari |

# Description

With a custom alerts index pattern, the Overview Home link doesn't work properly. The `rule.level` seems to apply the default alerts index pattern and throws an error because it doesn't exist.

image

The issue seems to be related to this:
https://github.com/wazuh/wazuh-dashboard-plugins/blob/v4.8.0-2.10.0/plugins/main/public/controllers/overview/components/last-alerts-stat/last-alerts-stat.tsx#L90

### PR DESCRIPTION
**Pull request by Desvelao**
### Description
This pull request fixes the hardcoded value of index pattern on the filters of home KPI links.

Side changes:
- Fix a problem in the home KPI links when using a index pattern with different title and id.
 
### Issues Resolved
#6776

### Evidence
image

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :black_circle: | :black_circle: | :black_circle: |
| With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by asteriscos** [comment]
| Test | Chrome | 
| --- |  --- |
| With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :black_circle: | 
| With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :black_circle: | 

**Details**
<details>
<summary>:black_circle: With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :black_circle:


</details>

<details>
<summary>:black_circle: With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :black_circle:


</details>

**Review by asteriscos — APPROVED** [review]
CR: :green_circle: 
Test: :green_circle:

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ All notable changes to the Wazuh app project will be documented in this file.
 
 - Support for Wazuh 4.8.1
 
+### Fixed
+
+- Fixed home KPI links with custom or index pattern whose title is different to the id [#6777](https://github.com/wazuh/wazuh-dashboard-plugins/pull/6777)
```
Comment (jbiset) [inline CHANGELOG.md:13]: Could it be that the `or` is not necessary?

**Review by jbiset — APPROVED** [review]
## CR: :green_circle: 

## Test UI :green_circle: 

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :green_circle: | :green_circle: | :black_circle: |
| With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :green_circle: | :green_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :green_circle:

image

image


Firefox - :green_circle: 

image

image

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :green_circle:

image

image


Firefox - :green_circle: 

image

image


Safari - :black_circle:

</details>

**Comment by Machi3mfl** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :green_circle: | :black_circle: | :black_circle: |
| With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors | :green_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: With an index pattern with the same title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :green_circle:

Screenshot 2024-06-19 at 12 25 34

Screenshot 2024-06-19 at 12 26 13


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: With an index pattern with a different title and ID, go to Home > Overview, the KPI links should display and redirect correctly to the Discover app and the filters should be applied without errors</summary>

Chrome - :green_circle:

Screenshot 2024-06-19 at 12 24 00

Screenshot 2024-06-19 at 12 24 46

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Pull request merged**